src/components/TextInput/BaseTextInput/index.native.tsx

```diff
@@ -216,48 +247,47 @@ function BaseTextInput(props) {
         }
     };
 
-    // eslint-disable-next-line react/forbid-foreign-prop-types
-    const inputProps = _.omit(props, _.keys(baseTextInputPropTypes.propTypes));
-    const hasLabel = Boolean(props.label.length);
-    const isReadOnly = _.isUndefined(props.readOnly) ? props.disabled : props.readOnly;
-    const inputHelpText = props.errorText || props.hint;
-    const placeholder = props.prefixCharacter || isFocused || !hasLabel || (hasLabel && props.forceActiveLabel) ? props.placeholder : null;
-    const maxHeight = StyleSheet.flatten(props.containerStyles).maxHeight;
-    const textInputContainerStyles = StyleSheet.flatten([
+    const hasLabel = Boolean(label?.length);
+    const isReadOnly = inputProps.readOnly ?? inputProps.disabled;
+    const inputHelpText = errorText || hint;
+    const placeholderValue = !!prefixCharacter || isFocused || !hasLabel || (hasLabel && forceActiveLabel) ? placeholder : undefined;
+    const maxHeight = StyleSheet.flatten(containerStyles)?.maxHeight;
+    const newTextInputContainerStyles: StyleProp<ViewStyle & FlexStyle> = StyleSheet.flatten([
         styles.textInputContainer,
-        ...props.textInputContainerStyles,
-        props.autoGrow && StyleUtils.getWidthStyle(textInputWidth),
-        !props.hideFocusedState && isFocused && styles.borderColorFocus,
-        (props.hasError || props.errorText) && styles.borderColorDanger,
-        props.autoGrowHeight && {scrollPaddingTop: 2 * maxHeight},
+        textInputContainerStyles,
+        autoGrow && StyleUtils.getWidthStyle(textInputWidth),
+        !hideFocusedState && isFocused && styles.borderColorFocus,
+        (!!hasError || !!errorText) && styles.borderColorDanger,
+        autoGrowHeight && {scrollPaddingTop: typeof maxHeight === 'number' ? 2 * maxHeight : undefined},
```

Comment:
Could you explain why are you doing this `typeof maxHeight === 'number'` in some places?

Reply:
yeah, `maxHeight` is type of `DimensionValue` which can be string so I need to check if this value is just a number to do this equation